### PR TITLE
feat: include proxy-wasm using standard cargo tooling

### PR DESCRIPTION
**Pull request by leonm1**
This fixes the bug described in https://github.com/leonm1/service-extensions-samples/commit/9f97727b05103cbf6e7fd9013daa263ffe65a37c, where multiple log crates can be linked into the resulting wasm binary, as well as allowing standard rust IDE tooling to find all of the included crates.

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -35,5 +36,7 @@ targets = [
     "x86_64-apple-darwin",
     "x86_64-pc-windows-msvc",
     "x86_64-unknown-linux-gnu",
+    "wasm32-unknown-unknown",
+    "wasm32-wasi",
```
Comment (martijneken) [inline Cargo.toml:40]: You can delete all the other targets; this is what ProxyWasm supports amd the user guide recommends

File: Cargo.lock [hunk 2]
```diff
@@ -0,0 +1,177 @@
+# This file is automatically @generated by Cargo.
```
Comment (martijneken) [inline Cargo.lock:1]: Dupe of cargo/Cargo.raze.lock?

Comment (leonm1) [reply Cargo.lock]: Done.

File: Cargo.toml (hunk 1 above)
Comment (leonm1) [reply Cargo.toml:37]: Done.

**Pull request merged**